components_library/processors/complex_generator_core.py

```diff
@@ -0,0 +1,116 @@
+from m5.util.convert import toLatency, toMemoryBandwidth
+
+from components_library.processors.cpu_types import CPUTypes
+from components_library.processors.py_generator_core import PyGeneratorCore
+
+
+from .py_generator_core import PyGeneratorCore
+
+
+class ComplexGeneratorCore(PyGeneratorCore):
```

Comment:
This needs tons of comments. I'm an expert at this stuff and I don't understand how I'm supposed to use any function in this file.